aqandu/gaussian_model_utils.py

```diff
@@ -149,36 +174,57 @@ def saveMatrixToFile(matrix, filename):
 # goal is to fill in zero/bad elements in between two values
 # only do short distances, e.g.  1 > <  SENSOR_INTERPOLATE_DISTANCE (missing bins)
 def interpolateBadElements(matrix, bad_value = 0):
-    row_index = 0
+    num_interp_fails = 0
     for row in matrix:
-        row_index += 1
         prevValueIndex = None
+        this_fail = False
         for i in range(row.shape[0]):
             if row[i] != bad_value:
                 if prevValueIndex is None:
                     prevValueIndex = i
+# this takes care of the boundary at the beginning of the time sequence
+                    if (i > 0) and (i < SENSOR_INTERPOLATE_DISTANCE):
```

Comment:
Looks like the indentation is off a little here
```suggestion
                    # this takes care of the boundary at the beginning of the time sequence
                    if (i > 0) and (i < SENSOR_INTERPOLATE_DISTANCE):
```